cicd/internal/flags/it-flags.go

```diff
@@ -84,6 +86,13 @@ func PrivateConnectivity() string {
 	return ""
 }
 
+func SpannerHost() string {
+	if dSpannerHost == "" {
+		return "-DspannerHost=" + "https://staging-wrenchworks.sandbox.googleapis.com/"
```

Comment:
Does this make the staging instance as the default host?

Also, I don't see the changes in the Github actions that will be required to pass the `spannerHost`, will you be doing that in a separate PR?

I don't understand how the same test is being executed for both a prod and the staging Spanner host.

Reply:
@manitgupta First of all, thanks for pointing that out because you made me realized I never actually call that method in the workflows (fixed).

Second of all, take the following test (part of this PR) as an example:
```
@Test
  public void testSpannerToGCSAvroStaging() throws IOException {
    spannerResourceManager =
        SpannerResourceManager.builder(testName, PROJECT, REGION, Dialect.GOOGLE_STANDARD_SQL)
            .maybeUseStaticInstance()
            .maybeUseCustomHost()
            .build();
    testSpannerToGCSAvroBase(
        paramAdder ->
            paramAdder.addParameter("spannerHost", spannerResourceManager.getSpannerHost()));
  }
```

Notice how it instantiates a separate resource manager instead of using a top-level one like the other tests. Also notice it calls `.maybeUseCustomHost()` - this method (defined in the PR) will use the custom host, if one is provided on the command line with `-DspannerHost=...`. 

We do the same thing for `maybeUseStaticInstance()` which was the ideology I was trying to follow as I like how it doesn't fail the test should a user run who can't access the endpoint (assuming the `-DspannerHost` flag was not passed).